### DISCUSSION TITLE
Jakarta Core profile

### DISCUSSION THREAD

**Discussion by Thihup** [piranhacloud/piranha#2581] [Ideas]
The new core profile will include fewer specifications:

- Jakarta Annotations 2.1
- Jakarta Contexts and Dependency Injection 4.0 (CDI Lite section)
- Jakarta Dependency Injection 2.0
- Jakarta Interceptors 2.1
- Jakarta JSON Processing 2.1
- Jakarta JSON Binding 3.0
- Jakarta RESTful Web Services 3.1

Should we try to get the Core profile certification before the Servlet certification?

**Comment by mnriem** [comment]
Go for it!

**Comment by arjantijms** [comment]
The core profile is but a few actual TCKs. Annotations doesn't have a TCK and neither does Interceptors (tested as part of CDI). 

JSON doesn't have a TCK an implementation has to pass if it used unmodified jars that have already passed the TCK (this is very weird, but not my rules).

So essentially we have to pass CDI, DI (both are for Weld actually in our case) and Jakarta REST.

To run the CDI TCK, see https://github.com/eclipse-ee4j/glassfish/tree/master/appserver/tests/tck/cdi
